src/operator/linalg_impl.h

```diff
@@ -103,8 +103,8 @@ void linalg_batch_gemm<cpu, DType>(const Tensor<cpu, 3, DType>& A, const Tensor<
 LINALG_CPU_GEMM(sgemm, float)
 LINALG_CPU_GEMM(dgemm, double)
 
-LINALG_CPU_BATCH_GEMM(float)
-LINALG_CPU_BATCH_GEMM(double)
+LINALG_XPU_BATCH_GEMM(cpu, float)
```

Comment:
This causes an error in the amalgamation build as there MSHADOW_USE_CBLAS==0 && MSHADOW_USE_MKL==0. As a consequence, the dummy stubs starting at line 84 will be generated, but they are named "CPU_GEMM" and not "XPU_GEMM". So you may have to change the names of these dummy stubs such that a call to LINALG_XPU_BATCH_GEMM(cpu,...)  correctly generates these stubs instead.

Reply:
Changed, let's see if it builds correctly.